storage/innobase/fil/fil0crypt.cc

```diff
@@ -118,18 +118,28 @@ uint get_global_default_encryption_key_id_value();
 
 #define DEBUG_KEYROTATION_THROTTLING 0
 
-uint fil_get_encrypt_info_size(const uint iv_len) {
-  return ENCRYPTION_MAGIC_SIZE + 2  //length of iv
-         + 4                        // space id
-         + 2                        // offset
-         + 1                        // type
-         + 4                        // min_key_version
-         + 4                        // key_id
-         + 1                        // encryption
-         + iv_len                   // iv
-         + 4                        // encryption rotation type
-         + ENCRYPTION_KEY_LEN       // tablespace key
-         + ENCRYPTION_KEY_LEN;      // tablespace iv
+/* Return the size of encryption header used Keyring key
+encryption.
+@return length of encryption header */
+static uint fil_get_keyring_encrypt_info_size() {
+  const uint size = ENCRYPTION_MAGIC_SIZE + 2  //length of iv
+                    + 4                        // space id
+                    + 2                        // offset
+                    + 1                        // type
+                    + 4                        // min_key_version
+                    + 4                        // key_id
+                    + 1                        // encryption
+                    + CRYPT_SCHEME_1_IV_LEN    // iv (16 bytes)
+                    + 4                        // encryption rotation type
+                    + ENCRYPTION_KEY_LEN       // tablespace key
+                    + ENCRYPTION_KEY_LEN;      // tablespace iv
+  /* The size of keyring key encrption header cannot cross the Master
+  Key header. This is because the bytes followed by MK header are used
+  by other features like SDI, encryption progress (MK) etc. If we cross
+  the MK encryption header size ENCRYPTION_INFO_MAX_SIZE, we corrupt the
+  header and other features will not work */
+  ut_a(size < ENCRYPTION_INFO_MAX_SIZE);
```

Comment:
static_assert, if both are constants, compile time checking is sufficient and efficient